### ISSUE TITLE
Brush Puck replaced by wrong cursor when leaving canvas area

### ISSUE THREAD

**Issue by redrodent** [drawpile/Drawpile#836]
iMac, up to date. Drawpile, up to date. When leaving the canvas area of the screen and going onto one of the windows with tools or brush options, etc. The painting puck gets replaced by either a regular mouse cursor, or the tools that let you adjust the size of your tool windows. The only workaround I've found is tabbing out, which is inconvenient and only fixes it for as long as I stay on the canvas.

Screenshot attached with a drawing of the issue since I can't screen cap my cursor....

Basically it looks like Drawpile forgets that I want to be using the drawing/painting puck when I go back onto the canvas from my toolbars/windows.

![Screen Shot 2019-11-21 at 8 50 24 PM](https://user-images.githubusercontent.com/10472724/69398784-8f7f3280-0ca0-11ea-879b-af1791886830.png)

Tabbing out and back into Drawpile fixes it, but not permanently.

**Comment by redrodent** [comment]
Taking my cursor far enough away from my Wacom Cintiq display also resets the cursor.

**Comment by Wade821** [comment]
Thanks for alerting us of the issue. When providing information about your operating system and drawpile version however, please provide specific numbers. We don't know if latest means the latest your computer can get because it's old, or if it's the newest release from Apple. 

Same with Drawpile. Some people update a month ago and don't actually check for updates. Oof. Giving numbers assures that we're on the same page.

**Comment by dustiibunnii** [comment]
Oh! This happens on my MacBook Pro as well. Currently I am using a Catalina 10.15.5, with drawpile 2.1.17, and whenever I go off screen, and then back in, the circle will be overlapped by the regular curser, or if I was typing in chat, by the ‘I’ symbol. It doesn’t appear to interfere with the strokes, it’s just the curser is changed.

**Comment by userwer** [comment]
I am experiencing the same issue. Whenever my cursor leaves the canvas area, it will stay in the cursor shape upon returning, and most of the time the brush/eraser outlines do not reappear. Sometimes (I am not sure exactly when) both the cursor and the outlines are there together. I am able to temporarily resolve the problem by clicking on a drop-down menu (like the edit menu, or the zoom menu where it lists percentages), and clicking directly into the canvas with my pen. The brush and eraser outlines will be back to normal until my cursor leaves the canvas again.

I'm using Void Linux on kernel 5.10.10_1, with Drawpile compiled from github; `git show` tells me I am on commit 4523f2d5ee279273aa79df6415bdb774dd147a22.

**Comment by askmeaboutlo0m** [comment]
I been having this issue too and implemented a workaround in #969 I think. It works for me on Linux now anyway, not sure if it solves the situation on Mac.
